alf/nest/nest.py

```diff
@@ -883,3 +883,36 @@ def set_field(nested, field, new_value):
     """
 
     return transform_nest(nested, field, lambda _: new_value)
+
+
+def transpose(nested, shallow_nest):
+    """Given a nest and its shallow nest structure ``B``, assuming that each leaf
```

Comment:
Is "shallow nest structure" of a nest a well-known concept? Not sure what's the best way to phrase this though, but it took me a while to understand that `shallow_nest` here defines "what are considered leaves".

Reply:
> Is "shallow nest structure" of a nest a well-known concept? Not sure what's the best way to phrase this though, but it took me a while to understand that `shallow_nest` here defines "what are considered leaves".

Actually `shallow_nest` is meant to be the top level of a nest. For example, `dict(x=1,y=1)` is a shallow nest of `dict(x=[0,1], y=1)`. This term has been used in `flatten_up_to` and `map_structure_up_to`. But for clarity, I've changed it to "top level".